Aikido.Zen.DotNetCore/Middleware/ContextMiddleware.cs

```diff
@@ -145,6 +145,86 @@ private bool TryPrepareContext(HttpContext httpContext, out ConcurrentDictionary
 
         }
 
+        /// <summary>
+        /// Flattens query parameters into individual dictionary entries with indexing for multiple values.
+        /// </summary>
+        /// <param name="query">The query collection</param>
+        /// <returns>A dictionary with flattened query parameters</returns>
+        private static IDictionary<string, string> FlattenQueryParameters(IQueryCollection query)
+        {
+            var result = new Dictionary<string, string>();
+
+            foreach (var kvp in query)
+            {
+                var values = kvp.Value;
+                if (values.Count == 1)
+                {
+                    result[kvp.Key] = values[0];
+                }
+                else
+                {
+                    for (int i = 0; i < values.Count; i++)
+                    {
+                        string dictKey = i == 0 ? kvp.Key : $"{kvp.Key}[{i}]";
```

Comment:
cool solution, but are we sure that the first element is the default one? usually last header/last query etc is the default? Would also add a comment with an example here